### PR TITLE
TDR-3244 View transfer correct series status

### PR DESCRIPTION
**Pull request by thanhz**
As suggested in the previous [PR](https://github.com/nationalarchives/tdr-transfer-frontend/pull/3078#discussion_r1231912019). This will now set the series to "InProgress' on first load. This also fixes the View transfer displaying _Contact Us_ instead of _In progress_ when users goes to select a series then clicks cancel.

Related api change pr: https://github.com/nationalarchives/tdr-consignment-api/pull/632

### CODE REVIEW

File: app/controllers/SeriesDetailsController.scala [hunk 1]
```diff
@@ -79,6 +79,9 @@ class SeriesDetailsController @Inject() (
               .uncache()
           )
         case _ =>
+          if (!seriesStatus.contains(InProgressValue.value)) {
```
Comment (vimleshtna) [inline app/controllers/SeriesDetailsController.scala:82]: Looks good, can you please add a unit test to verify if the `seriesStatus` is `None` then it should add a consignment status and if it is in progress then it shouldn't.

**Review by vimleshtna — COMMENTED** [review]
Thanks for doing this. Looks good, just asked to add a couple of unit tests, thanks.

File: app/controllers/ViewTransfersController.scala [hunk 2]
```diff
@@ -96,8 +96,8 @@ class ViewTransfersController @Inject() (
       case s if s.containsStatuses(ClientChecksType, UploadType) => toClientSideChecksAction(statuses, consignmentId, judgmentType)
       case s if s.containsStatuses(TransferAgreementType) =>
         toTransferAgreementAction(s.find(_.statusType == TransferAgreementType.id).get, consignmentId)
-      case s if s.statusValue(SeriesType).contains(CompletedValue.value) =>
-        UserAction(InProgress.value, routes.TransferAgreementPart1Controller.transferAgreement(consignmentId).url, Resume.value)
+      case s if s.containsStatuses(SeriesType) =>
```
[the remark this answers is not in the document]
Comment (thanhz) [reply app/controllers/ViewTransfersController.scala:99]: Thanks, I forgot about this

File: app/controllers/SeriesDetailsController.scala (hunk 1 above)
Comment (thanhz) [reply app/controllers/SeriesDetailsController.scala]: As suggested I've added unit tests to cover those scenarios. I had to move the logic into its own function to make the testing a bit more manageable.

File: app/controllers/SeriesDetailsController.scala [hunk 3]
```diff
@@ -97,6 +95,12 @@ class SeriesDetailsController @Inject() (
     }
     DropdownField(form("series").id, "", "", "", Nil, multiValue = false, options, None, isRequired = true, errors.toList)
   }
+
+  def addSeriesStatus(seriesStatus: Option[String], consignmentId: UUID, request: Request[AnyContent]) = {
```
Comment (vimleshtna) [inline app/controllers/SeriesDetailsController.scala:99]: Sorry, just a small thing, actually, if we are not using any method outside of the class then we should make it private but I think you can still test it with the main method.

**Review by vimleshtna — COMMENTED** [review]
Looks good to me, just a small suggestion.

File: app/controllers/SeriesDetailsController.scala (hunk 3 above)
Comment (thanhz) [reply app/controllers/SeriesDetailsController.scala]: No worries. I've moved the method back into the main method and made changes to the tests.

**Review by vimleshtna — APPROVED** [review]
Looks good, thanks
